fri/src/verifier.rs

```diff
@@ -0,0 +1,282 @@
+use std::error::Error;
+
+use ark_ff::{FftField, PrimeField};
+use ark_poly::{domain::EvaluationDomain, Radix2EvaluationDomain};
+use randomness::Prng;
+use sha3::Digest;
+
+use crate::{lde::MultiplicativeLDE, parameters::FriParameters};
+use channel::{fs_verifier_channel::FSVerifierChannel, Channel, VerifierChannel};
+use commitment_scheme::{
+    make_commitment_scheme_verifier, table_verifier::TableVerifier, CommitmentHashes,
+};
+
+#[allow(dead_code)]
+pub type FirstLayerQueriesCallback<F> = fn(&[u64]) -> Vec<F>;
```

Comment:
What does this do? 

Might be cleaner to just add another trait parameter to the `FriVerifier`